File: solutions/1750. Minimum Length of String After Deleting Similar Ends/1750.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def minimumLength(self, s: str) -> int:
    left = 0
    right = len(s) - 1

    while left < right and s[left] == s[right]:
      c = s[left]
      while left <= right and s[left] == c:
        left += 1
      while left < right and s[right] == c:
        right -= 1

    return right - left + 1

File: solutions/1750. Minimum Length of String After Deleting Similar Ends/1750.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
public:
  int minimumLength(string_view s) {
    int left = 0;
    int right = s.length() - 1;

    while (left < right && s[left] == s[right]) {
      char const c = s[left];

      while (left <= right && s[left] == c)
        ++left;

      while (left < right && s[right] == c)
        --right;  
      }

    return right - left + 1;
  }
};